oat-sa/extension-tao-itemqti
# Feature/TR-5015/Apply the text converter

### Related to: https://oat-sa.atlassian.net/browse/TR-5015

### Requires

- [ ] https://github.com/oat-sa/tao-core-sdk-fe/pull/163
- [ ] https://github.com/oat-sa/tao-core/pull/3710

### Summary

Convert ambiguous symbols to plain ASCII when setting the correct response of both TextEntry and ExtendedTextEntry interactions.

### Details

To share the common behavior, a helper has been added: `'taoQtiItem/qtiCreator/widgets/helpers/stringResponse'`.
It handles the reading and the writing of string values for the correct answer of a response declaration.

Note: unfortunately, I had to keep lodash in the helper as the spread operator in object is not yet supported by our bundler.

### How to test
- checkout the branch on `taoQtiItem` and `tao`
- install the dependencies on `tao/views` (`cd tao/views && npm ci`)
- run the unit tests (`cd tao/views/build && npx grunt connect:dev taoqtiitemtest`, or open your browser at `https://HOSTNAME/taoQtiItem/views/js/test/qtiCreator/widgets/helpers/stringResponse/test.html`)
- create an item and add TextEntry/ExtendedTextEntry interactions (you can also import the item shared with the ticket)
- set the correct response, adding full-width numbers in it (ex: `１２３４５６７８９０`), and check that the response is registered with plain ASCII number instead
